analysis/filter/EdgeFilter/src/main/kotlin/de/maibornwolff/codecharta/filter/edgefilter/EdgeProjectBuilder.kt

```diff
@@ -116,16 +119,16 @@ class EdgeProjectBuilder(private val project: Project, private val pathSeparator
     }
 
     private fun getAggregatedAttributes(
-        listOfAttributes: MutableList<String>,
-        filteredEdges: List<Edge>
+            listOfAttributes: MutableList<String>,
+            filteredEdges: List<Edge>
     ): MutableMap<String, Any> {
         val aggregatedAttributes: MutableMap<String, Any> = mutableMapOf()
 
         listOfAttributes.forEach { key: String ->
             val attributeType = getAttributeTypeByKey(key)
             val filteredAttribute = filteredEdges.filter { edge: Edge -> edge.attributes.containsKey(key) }
             var aggregatedAttributeValue =
-                filteredAttribute.sumBy { edge: Edge -> edge.attributes[key].toString().toFloat().toInt() }
+                    filteredAttribute.sumBy { edge: Edge -> edge.attributes[key].toString().toFloat().toInt() }
```

Comment:
What on earth is this conversion? :D I know this is old code, but we should work on that.
Is that not working?
```suggestion
                    filteredAttribute.sumBy { edge: Edge -> edge.attributes[key].toInt() }
```

Reply:
Thanks for the note @NearW 😄
I finally got around to investigating that. Here is what I discovered:
- *toString()* is needed because `edge.attributes[key]` is of type `any`. So `toFloat()` and `toInt()` can't cast it directly.
- *toFloat()* is needed because the result at that point would be "90.0" and to `toInt()` throws an error when trying to cast it. I assume because of the ".".
So casting to a string avoids the problem with `any` and casting to a float avoids the problem with `.`.

However, I fixed the deprecation problem because of `sumBy` and converted to `sumOf`. So that's an improvement at least 👍